### ISSUE TITLE
Error - using an integer in a method of a class

### ISSUE THREAD

**Issue by r-k-o** [petelomax/Phix#2]
the code below does not compile it gives

`
D:\Phix\pmain.e:8294 in procedure Assignment()
attempt to divide by 0
    tidx = 1055
    Type = 1004
    CompoundAssignment = 1
    subscript = 0
    sstokline = 112'p'
    sstokcol = 2581
    noofsubscripts = 1
    wasMapEndToMinusOne = 1
    savetokline = <novalue>
    savetokcol = <novalue>
    eqline = 112'p'
    eqcol = 2589
    osi = <novalue>
    tvar = <novalue>
    w = <novalue>
    symtabN = {}
    snNtyp = 3
    state = 65811
    lprev = <novalue>
    rtype = 15
    statemod = <novalue>
    LHStype = <novalue>
    idii = <novalue>
    wasdot = 1
    wasfromsubss = 0
    struct_fields = {{1055,`line_no`}}
    pstype = 2
    petype = 1004
    etype = 1
    bStruct = 0
    fN = <novalue>
    field_name = `line_no`
    fN = <novalue>
    p1 = <novalue>
    p2 = <novalue>
    opcode = <novalue>
    i = <novalue>
    i = <novalue>
    fN = <novalue>
    rep = <novalue>
    j = <novalue>
    j = <novalue>
    i = <novalue>
    j = <novalue>
    j = <novalue>
... called from D:\Phix\pmain.e:12170 in procedure Statement()
    Type = 1004
    N = 1055
    field_name = `line_no`
    p1 = <novalue>
    p2 = <novalue>
    opcode = <novalue>
... called from D:\Phix\pmain.e:12329 in procedure Block()
    flags = 0
    drop_scope = 0
... called from D:\Phix\pmain.e:9621 in procedure DoWhile()
    wasEmit = 1
    emitBlock = 1
    saveIchain = -1
    saveExitBP = -1
    saveContinueBP = -1
    tmp = 1
    loopTop = 55'7'
    sytmp = {6376,1,0,20993,0,0,1,1,39''',0,5756,1,0}
    wasSideEffects = 0
    waslMask = 114702
    thispt = 52'4'
... called from D:\Phix\pmain.e:12139 in procedure Statement()
    Type = <novalue>
    N = <novalue>
    field_name = <novalue>
    p1 = <novalue>
    p2 = <novalue>
    opcode = <novalue>
... called from D:\Phix\pmain.e:12329 in procedure Block()
    flags = 0
    drop_scope = 0
... called from D:\Phix\pmain.e:9372 in procedure DoIf()
    emitElse = 1
    wasEmit = 1
    ifBP = 41')'
    saveIchain = -1
    EndIfBP = 0
    tmp = 1056
    sytmp = <novalue>
    elsevalid = 2
    wasSideEffects = 0
    plain = <novalue>
    testfor = <novalue>
    iftop = 11
    ctrlink = 11
    ctrltyp = <novalue>
    scode = <novalue>
    wasEmit2 = <novalue>
... called from D:\Phix\pmain.e:12137 in procedure Statement()
    Type = <novalue>
    N = <novalue>
  ......
`
If you take out:  line_no += 1 it will compile - see below, it is marked.

`
constant EOF       = -1
constant DEFAULTCF = 90
constant etx       = 0x3
constant ERR       = -#4000000000000000
constant WARNING   = ERR+1
constant NOERR     = 0

constant NPROMPT   = "NPROMPT"
constant PROMPT    = "PROMPT"
constant ENDPROMPT = "ENDPROMPT"
constant TRANS     = "TRANS"
constant ENDTRANS  = "ENDTRANS"

constant IF        = "IF"
constant ENDIF     = "ENDIF"
constant IS        = "IS"
constant FUNCTION  = "FUNCTION"
constant PROCEDURE = "PROCEDURE"
constant THEN      = "THEN"
constant AND       = "AND"
constant OR        = "OR"
constant ANDNOT    = "ANDNOT"
constant ORNOT     = "ORNOT"
constant CF        = "CF:"

public enum
  VARIABLE,
  STRING,
  NUMBER,
  OTHER

public enum  --- use in attributes for indexing
  ATTRNAME,
  ATTRTYP,
  ATTRPROMPT,
  ATTRTRANS,
  ATTRCF

--CONNECTOR 
public enum
  NONE,
  CAND = 1000,--1000
  CANDNOT,    --1001
  COR,        --1002
  CORNOT,     --1003
  PROC = 50,
  FUNC = 60

class parser

  integer line_no = 0
  integer ofile = -1
  object attr_list = {}
  object error_list = {}

  function removeDuplicateNumbers(sequence arr)
    integer n = length(arr)
    if n=0 then return {}  end if
    if n=1 then return arr end if
    arr = sort(arr)
    sequence temp={}
    -- Start traversing elements 
    for i=1 to n-1 do
      -- If current element is not equal to next element then store that current element 
      if arr[i] != arr[i+1] then
        temp &= arr[i]
      end if
    end for
    -- Store the last element as whether it is unique or repeated, it hasn't stored previously 
    temp &= arr[n]
    return temp
  end function

  function removeDuplicateNumbersPairs(sequence arr)
    integer n = length(arr)
    if n=0 then return {}  end if
    if n=1 then return arr end if
    arr = sort(arr)
    sequence temp={}
    -- Start traversing elements 
    for i=1 to n-1 do
      -- If current element is not equal to next element then store that current element 
      if arr[i] != arr[i+1] then
        temp &= {arr[i]}
      end if
    end for
    -- Store the last element as whether it is unique or repeated, it hasn't stored previously 
    temp &= {arr[n]}
    return temp
  end function

  function OpenRules(object filename)
  ofile = open(filename, "r")
    if ofile = -1 then
      return false
    end if
    return true
  end function

  procedure CloseRules()
    close(ofile)
  end procedure

  function ReadRulesLine()
    if ofile != EOF then
      while true do
        line_no += 1  -- <=========================Take this out and it compiles
        object line = gets(ofile)
        if line = -1 then
          return "" & etx -- EOF is returned at end of file
        end if
        if line[$] = '\n' then
          line = trim(line[1..$-1])
        end if      
        if (line != "") and (line[1..2] != "--") then
          return line
        end if
      end while
    end if
    return "" & etx -- EOF is returned at end of file
  end function
end class

--parser p = new()
`

**Comment by petelomax** [comment]
Yup, this is on my todo list, as per the end of http://phix.x10.mx/docs/html/structs.htm (the example I gave there was s.age += 1)

As a quick work-around, 'line_no = line_no +1' should work.

**Comment by r-k-o** [comment]
Thanks for the help.
the following crashes also:
error_list = error_list & {ERR, "Failed on prompt or trans", line_no}

line 8294 is in what file?

**Comment by petelomax** [comment]
Line 8294 is in file D:\Phix\pmain.e

**Comment by r-k-o** [comment]
thanks for the answer.
the following also says not implemented:
`
attr_list[at][ATTRPROMPT] &= lines

                          <==
`
seems like the indexing is not allowed - yet.
